src/components/ThreejsEditorModal.js

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+import * as THREE from "three";
+import {ModalBody} from "react-bootstrap";
+
+import {ShowIf} from "./ShowIf";
+import settings from "../settings";
+import ModalDialog from "./ModalDialog";
+import {materialsToThreeDSceneData} from "../utils";
+import {LoadingIndicator} from "./LoadingIndicator";
+
+import {THREE_D_BASE_URL, THREE_D_SOURCES} from "../enums";
+
+export class ThreejsEditorModal extends ModalDialog {
+
+    constructor(props) {
+        super(props);
+        window.THREE = THREE;
+        this.domElement = undefined;
+        this.state = {areScriptsLoaded: false};
+        this.injectScripts();
+    }
+
+    componentDidUpdate(prevProps, prevState, snapshot) {
+        window.localStorage.removeItem("threejs-editor");
+    }
+
+    injectScripts() {
+        THREE_D_SOURCES.forEach(src => {
```

Comment:
The logic used inside this function needs a fair amount of explanation, perhaps in the docstring and/or inline comments

Reply:
Done.